frontend/admin/src/js/components/apiManagedTable/SearchForm.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import { useIntl } from "react-intl";
+
+import DropdownMenu, {
+  MenuButton,
+  MenuItem,
+  MenuList,
+} from "@common/components/DropdownMenu";
+
+import { useDebounce, type SearchColumn } from "./basicTableHelpers";
+
+export interface SearchFormProps {
+  onChange: (val: string | undefined, col: string | undefined) => void;
+  searchBy?: Array<SearchColumn>;
+}
+
+const SearchForm: React.FC<SearchFormProps> = ({ onChange, searchBy }) => {
+  const intl = useIntl();
+  const [searchTerm, setSearchTerm] = React.useState<string | undefined>(
+    undefined,
+  );
+  const [column, setColumn] = React.useState<SearchColumn | undefined>(
+    undefined,
+  );
+  const debouncedSearchTerm = useDebounce(searchTerm, 200);
+  const showDropdown = searchBy && searchBy.length;
+
+  React.useEffect(() => {
+    onChange(debouncedSearchTerm, column?.value);
+    // NOTE: onChange causes infinite rerenders
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Ooh, this doesn't smell good.  Are we able to avoid disabling this?  Let me know if you'd like me to take a swing at it.

Reply:
Okay, it's fairly ugly but I was able to avoid using `useEffect` at all which is recommended.

Fixed with [remove need to useEffect to trigger onChange](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3260/commits/d1f64aff92c0c98b95bef0338da89915941fd30d)